sdk/metric/meter_test.go

```diff
@@ -2064,3 +2068,208 @@ func TestHistogramBucketPrecedenceOrdering(t *testing.T) {
+		},
+		{
+			name: "drop float64 observable",
+			views: []View{
+				func(i Instrument) (Stream, bool) {
+					if strings.HasPrefix(i.Name, "observable.float64") {
+						return Stream{Aggregation: AggregationDrop{}}, true
+					}
+					return Stream{}, false
+				},
+			},
+			expectedObservableName: []string{
+				"observable.int64.counter",
+				"observable.int64.up.down.counter",
+				"observable.int64.gauge",
+			},
+			expectedLogs: []map[string]interface{}{
+				{
+					"error":  "observable instrument not registered for callback",
+					"msg":    "failed to record",
+					"name":   "unregistered.observable.int64.counter",
+					"number": "int64",
+				},
+				{
+					"error":  "observable instrument not registered for callback",
+					"msg":    "failed to record",
+					"name":   "unregistered.observable.float64.counter",
+					"number": "float64",
+				},
+			},
+		},
+		{
+			name: "drop int64 observable",
+			views: []View{
+				func(i Instrument) (Stream, bool) {
+					if strings.HasPrefix(i.Name, "observable.int64") {
+						return Stream{Aggregation: AggregationDrop{}}, true
+					}
+					return Stream{}, false
+				},
+			},
+			expectedObservableName: []string{
+				"observable.float64.counter",
+				"observable.float64.up.down.counter",
+				"observable.float64.gauge",
+			},
+			expectedLogs: []map[string]interface{}{
+				{
+					"error":  "observable instrument not registered for callback",
+					"msg":    "failed to record",
+					"name":   "unregistered.observable.int64.counter",
+					"number": "int64",
+				},
+				{
+					"error":  "observable instrument not registered for callback",
+					"msg":    "failed to record",
+					"name":   "unregistered.observable.float64.counter",
+					"number": "float64",
+				},
+			},
+		},
+		{
+			name: "drop unregistered observable",
+			views: []View{
+				func(i Instrument) (Stream, bool) {
+					if strings.HasPrefix(i.Name, "unregistered.observable") {
+						return Stream{Aggregation: AggregationDrop{}}, true
+					}
+					return Stream{}, false
+				},
+			},
+			expectedObservableName: []string{
+				"observable.int64.counter",
+				"observable.int64.up.down.counter",
+				"observable.int64.gauge",
+				"observable.float64.counter",
+				"observable.float64.up.down.counter",
+				"observable.float64.gauge",
+			},
+			expectedLogs: []map[string]interface{}{},
+		},
+	}
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			logEntries := []map[string]interface{}{}
+			global.SetLogger(
+				funcr.NewJSON(
+					func(obj string) {
+						var entry map[string]interface{}
+						_ = json.Unmarshal([]byte(obj), &entry)
+						logEntries = append(logEntries, entry)
+					},
+					funcr.Options{Verbosity: 0},
+				),
+			)
+
+			reader := NewManualReader()
+			meter := NewMeterProvider(WithView(tt.views...), WithReader(reader)).Meter("TestObservableDropAggregation")
+
+			aiCounter, err := meter.Int64ObservableCounter("observable.int64.counter")
```

Comment:
nit: can we rename the names from `aiCounter` to eg. `intCnt`. The `ai` prefix is cryptic while `cnt` is rather a common shorthand for `counter`. `hist` can be used for Histogram. `udcnt` for UpDownCounter